client/src/livingdocs-component-app/app.js

```diff
@@ -0,0 +1,314 @@
+    try {
+      const targets = await this.qTargets.get('availableTargets');
+      this.target = targets.filter(target => {
+        return target.key === this.targetKey;
+      })[0];
+    } catch (e) {
+      // nevermind, we'll check if target exists before loading preview
+    }
+  }
+
+  loadView() {
+    if (this.selectedItem) {
+      this.title = this.items
+        .filter(item => {
+          return item.id === this.selectedItem.id;
+        })
+        .map(item => {
+          return item.conf.title;
+        })[0];
+
+      if (this.target && this.selectedItem) {
+        this.loadPreview();
+      }
+      this.loadDisplayOptions();
+    }
+  }
+
+  selectItem(item) {
+    this.loadSelectedItem(item);
+    this.loadView();
+  }
+
+  loadSelectedItem(item) {
+    this.title = item.conf.title;
+    const alreadySelected = this.selectedItems.filter(selectedItem => {
+      return item.conf._id === selectedItem.id;
+    });
+
+    if (alreadySelected.length > 0) {
+      this.selectedItem = alreadySelected[0];
+    } else {
+      this.selectedItem = {
+        id: item.conf._id,
+        toolRuntimeConfig: {
+          displayOptions: {}
+        }
+      };
+    }
+  }
+
+  changeDisplayOptions() {
+    this.selectedItems = this.selectedItems.filter(item => {
+      return item.id !== this.selectedItem.id;
+    });
+    this.selectedItems.push(this.selectedItem);
+    this.loadPreview();
+  }
+
+  insertItem() {
+    // delete all displayOptions set to false
+    let displayOptions = this.selectedItem.toolRuntimeConfig.displayOptions;
+    Object.keys(displayOptions).forEach(displayOption => {
+      if (!displayOptions[displayOption]) {
+        delete displayOptions[displayOption];
+      }
+    });
+
+    const message = {
+      action: 'update',
+      params: this.selectedItem
+    };
+    window.parent.postMessage(message, '*');
+  }
+
+  async reloadItems() {
+    this.items = [];
+    const tools = await this.getTools();
+    const result = await this.loadItems(this.currentSearchString);
+
+    this.items =  result.items.map(item => {
+      let itemTool = tools
+        .filter(tool => {
+          return tool.name === item.getToolName();
+        })[0];
+
+      if (itemTool) {
+        item.conf.icon = itemTool.icon;
+      }
+
+      return item;
+    });
+
+    this.bookmark = result.bookmark;
+    this.updateMoreItemsAvailableState(result);
+
+    return this.items;
+  }
+
+  async loadItems(searchString, bookmark) {
+    try {
+      this.itemsLoading = true;
+      let numberOfItemsToLoadPerStep = 6;
+
+      const result = await this.itemStore.getItems(searchString, numberOfItemsToLoadPerStep, undefined, bookmark);
+      this.itemsLoading = false;
+      return result;
+    } catch (e) {
+      this.messageService.push('error', 'failedLoadingItems');
+    }
+  }
+
+  async loadMore() {
+    const result = await this.loadItems(this.currentSearchString, this.bookmark);
+    const tools = await this.getTools();
```

Comment:
this reloads the tools on every loadMore. I'd load them once in `attached`. or is there a reason why they should get reloaded here that I miss?